Army-Java/Ability/HealerAbility.java

```diff
@@ -0,0 +1,24 @@
+package com.gymfox.Army.Ability;
+
+import com.gymfox.Army.Exception.UnitIsDeadException;
+import com.gymfox.Army.Spells.Spell;
+import com.gymfox.Army.Units.Unit;
+
+public class HealerAbility extends Ability {
+    public HealerAbility(Unit currentUnit) {
+        super(currentUnit);
+    }
+
+    public void castSpell(Unit victim, Spell currentSpell) throws UnitIsDeadException {
+        if ( currentSpell.getSpellsType() == Spell.SpellsType.BATTLESPELL ) {
+            victim.takeMagicDamage(currentSpell.getHitPoints() / 2);
+
+            return;
+        }
+        if ( currentSpell.getSpellsType() == Spell.SpellsType.HEALSPELL) {
```

Comment:
So I'd probably implement logic of spell inside of spell.
Spell could have method `apply(Unit target)` when all logic will be implemented. So you wouldn't check spell type each time you use a spell